WebApp/webapp/src/pages/Prediction/index.tsx

```diff
@@ -1,15 +1,207 @@
-                Prediction feature
+
+  const [formVisible, setFormVisible] = useState(true)
+  const [progressVisible, setProgressVisible] = useState(false)
+  const [errorVisible, setErrorVisible] = useState(false)
+  const [successVisible, setSuccessVisible] = useState(false)
+  const [team1, setTeam1] = useState("Waiting On Team 1 Choice")
+  const [team2, setTeam2] = useState("Waiting On Team 2 Choice")
+  const [errMessage, setErrorMessage] = useState("Error")
+  const [winner, setData] = useState("")
+  const [winnerImage, setWinImg] = useState("")
+  const [winProb, setWinProb] = useState(0)
+  const [team1image, setTeam1Image] = useState("nba-logo.png")
+  const [team2image, setTeam2Image] = useState("nba-logo.png")
+  const [imageClass, setImageClass] = useState(styles.nothing)
+
+  const handleRefresh = () => {
+    window.location.reload();
+  }
+
+  function changeTeam1(e: ChangeEvent<HTMLSelectElement>) {
+    let newTeam: string = e.target.value;
+    let newTeamImage: string = newTeam.replace(/ /g,"_")
+    newTeamImage = newTeamImage.concat('.png')
+    setTeam1(newTeam)
+    setTeam1Image(newTeamImage)
+    if(newTeam === ""){
+      setTeam1Image("nba-logo.png")
+      setTeam1("Waiting On Team 1 Choice")
+    }
+  }
+
+  function changeTeam2(e: ChangeEvent<HTMLSelectElement>) {
+    let newTeam: string = e.target.value;
+    let newTeamImage: string = newTeam.replace(/ /g,"_")
+    newTeamImage = newTeamImage.concat('.png')
+    setTeam2(newTeam)
+    setTeam2Image(newTeamImage)
+    if(newTeam === ""){
+      setTeam2Image("nba-logo.png")
+      setTeam2("Waiting On Team 2 Choice")
+    }
+  }
+
+  function onSubmit() {
+    if (!(team1 !== "" && team2 !== "")) {
+      throwEmptyInputError();
+    } else if (team1 === team2) {
+      throwDuplicateTeamError();
+    } else {
+      getUncachedData();
+    }
+  }
+
+  function decimalToPercent(num: number): number{
+    return Math.trunc(num * 100) / 100;
+  }
+
+  function getUncachedData() {
+    setFormVisible(false)
+    setProgressVisible(true)
+    setErrorVisible(false)
+    axios.post(process.env.REACT_APP_API_URL + pred_LR, {
+      team1: team1,
+      team2: team2,
+      year: "2023"
+    })
+    .then((response) => {
+      setProgressVisible(false)
+      setFormVisible(false)
+      setSuccessVisible(true)
+      setErrorVisible(false)
+      if(response.data.prob_win >= 0.50){
+        setData(team1)
+        setWinImg(team1image)
+        setWinProb(decimalToPercent(response.data.prob_win))
+      }
+      else{
+        setData(team2)
+        setWinImg(team2image)
+        setWinProb(decimalToPercent(response.data.prob_loss))
+      }
+      setImageClass(styles.teamName)
+    })
+    .catch((error) => {
+      console.log("catch error")
+      setProgressVisible(false)
+      setErrorVisible(true)
+      let errorMessage = "Error retrieving from Server\n"
+      setErrorMessage(errorMessage.concat(error))
+    })
+  }
+
+  function throwEmptyInputError() {
```

Comment:
This isn't being called right. If you don't select any teams or only one team, it still allows form submission which just means that the api is called but returns a 400 error